forloop_modules/function_handlers/database_handlers.py

```diff
@@ -141,6 +142,8 @@ def direct_execute(self, dbtable_name, query, new_var_name):
                             variable_handler.new_variable(new_var_name, df_new)
 
                             ncrb.update_last_active_dataframe_node_uid(node_uid)
+                        else:
+                            raise HTTPException(status_code=response.status_code, detail="Error requesting new node from api")
```

Comment:
Why do we raise a `HTTPException` inside of a function handler? As FastAPI exception, when it's raised inside of the FastAPI endpoint, a corresponding response (to that error) will be automatically constructed by the webserver. Here, this Exception will bubble up through `Pipeline signal manager` or `ExecutionCore` acting as any other Exception, yet additionally obscuring the real reason for crash and misleading with it's name.

For cases like this, we have `CriticalPipelineError`, which can be accessed through `errors.errors` in `forloop_modules`.

Reply:
Instead of 
```
                        response = ncrb.new_node(pos=[500, 300], typ="DataFrame", fields=fields)
                        if response.status_code in [200, 201]:
                            (...)
                        else:
                            raise HTTPException(status_code=response.status_code, detail="Error requesting new node from api")
```
we can use:
```
                        response = ncrb.new_node(pos=[500, 300], typ="DataFrame", fields=fields)
                        response.raise_for_status()
                        (...)
```